### PR TITLE
コントローラのテスト

### PR DESCRIPTION
**Pull request by tomoaki0609**
# WHAT
・メッセージ一覧ページを表示するアクション

・ログインしている場合
・アクション内で定義している・インスタンス変数があるか
・該当するビューが描画されているか
・ログインしていない場合
・意図したビューにリダイレクトできているか
・メッセージを作成するアクション

・ログインしているかつ、保存に成功した場合
・メッセージの保存はできたのか
・意図した画面に遷移しているか
・ログインしているが、保存に失敗した場合
・メッセージの保存は行われなかったか
・意図したビューが描画されているか
・ログインしていない場合
・意図した画面にリダイレクトできているか

# WHY
コントローラが正常に動くため

### CODE REVIEW

**Comment by tomoaki0609** [comment]
doujoutomoaki-no-MacBook-Air:chat-space Doba$ bundle exec rspec spec/controllers/messages_controller_spec.rb
DEPRECATION WARNING: [Devise] including `Devise::TestHelpers` is deprecated and will be removed from Devise.
For controller tests, please include `Devise::Test::ControllerHelpers` instead.
 (called from <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:3)

MessagesController
  GET #index
    login
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
      assigns the requested message to @message
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
      assigns the requested group to @group
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
      assigns the requested messages to @messages
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
      assigns the requested groups to @groups
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
      renders the :index template
    not login
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
DEPRECATION WARNING: Using positional arguments in functional tests has been deprecated,
in favor of keyword arguments, and will be removed in Rails 5.1.

Deprecated style:
get :show, { id: 1 }, nil, { notice: "This is a flash message" }

New keyword style:
get :show, params: { id: 1 }, flash: { notice: "This is a flash message" },
  session: nil # Can safely be omitted.
 (called from block (3 levels) in <top (required)> at /Users/Doba/projects/chat-space/spec/controllers/messages_controller_spec.rb:20)
      renders the :index template
  POST #create
    user loged-in and successfully saved
      saves the new message in the database
      redirects to messages#index
    user loged-in but missed saved
      unsave the new message in the database
      redirects to group_messages_path
    not login
      redirects to sign-in page

Deprecation Warnings:

Using `stub` from rspec-mocks' old `:should` syntax without explicitly enabling the syntax is deprecated. Use the new `:expect` syntax or explicitly enable `:should` instead. Called from /Users/Doba/projects/chat-space/spec/support/controller_macros.rb:8:in `login_user'.


If you need more of the backtrace for any of these deprecations to
identify where to make the necessary changes, you can configure
`config.raise_errors_for_deprecations!`, and it will turn the
deprecation warnings into errors, giving you the full backtrace.

1 deprecation warning total

Finished in 0.9143 seconds (files took 19.8 seconds to load)
11 examples, 0 failures

File: spec/controllers/messages_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      it "renders the :index template" do
+        expect(response).to render_template :index
+      end
+
+    end
+
+    context 'not login' do
+
+      it "renders the :index template" do
+        expect(response).to render_template :index
```
Comment (taketnk) [inline spec/controllers/messages_controller_spec.rb]: ログインしていない時は、:indexにいけましたか？

**Comment by taketnk** [comment]
DEPRECATION WARNINGがでていますね。エラーではないのですが、今後使用できなくなるようなので、解消しておきましょう。解消法は、ググればだいたい出てくると思います。

**Comment by tomoaki0609** [comment]
default

**Comment by tomoaki0609** [comment]
default

**Comment by yamamotokeishiro** [comment]
よく出来ています！
しかし、DEPRECATION WARNINGがでている箇所の記述の意味は確認できていますでしょうか？

**Pull request merged**